app/graphql/collection_loader.rb

```diff
@@ -10,8 +10,11 @@ def initialize(model, association_name, scope = nil)
 
   def load(record)
     raise TypeError, "#{model} loader can't load association for #{record.class}" unless record.is_a?(model)
-    return Promise.resolve(read_association(record)) if association_loaded?(record)
-    super
+    if association_loaded?(record) && (!base_scope || base_scope.empty_scope?)
```

Comment:
`(!base_scope || base_scope.empty_scope?)` was added because we can use the default loader if the scope is empty anyway or the model does not support authorization so we don't have to take special care?

Reply:
it's actually both :slightly_smiling_face:
Main idea is, if this condition is true, we don't scope the preloaded records and the default preload is fine, that will probably only occur, if the record does not support authorization.
The chance for it to happen is very minimal though, so I would be ok deleting this whole branch.